cob_twist_controller/src/damping_methods/damping.cpp

```diff
@@ -118,20 +127,50 @@ double DampingManipulability::getDampingFactor(const Eigen::VectorXd& sorted_sin
 /**
  * Method returns the damping factor according to the least singular value.
  */
-double DampingLeastSingularValues::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
+Eigen::MatrixXd DampingLeastSingularValues::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
                                                     const Eigen::MatrixXd& jacobian_data) const
 {
     // Formula 15 Singularity-robust Task-priority Redundandancy Resolution
     double least_singular_value = sorted_singular_values(sorted_singular_values.rows() - 1);
+    uint32_t rows = sorted_singular_values.rows();
+    Eigen::MatrixXd lambda = Eigen::MatrixXd::Identity(rows,rows);
     if (least_singular_value < this->params_.eps_damping)
     {
         double lambda_quad = pow(this->params_.lambda_max, 2.0);
-        return sqrt( (1.0 - pow(least_singular_value / this->params_.eps_damping, 2.0)) * lambda_quad);
+        //lambda(sorted_singular_values.rows() - 1,sorted_singular_values.rows() - 1)= sqrt( (1.0 - pow(least_singular_value / this->params_.eps_damping, 2.0)) * lambda_quad);
+        //The value was powered squared later so the sqrt was removed
+        lambda = Eigen::MatrixXd::Identity(rows,rows)* (1.0 - pow(least_singular_value / this->params_.eps_damping, 2.0)) * lambda_quad;
+
+        return lambda;
     }
     else
     {
-        return 0.0;
+        return lambda.setZero();
     }
 }
-/* END DampingLeastSingularValues ************************************************************************************/
+/* END LeastSingularValues ************************************************************************************/
 
+/* BEGIN DampingSigmoid **********************************************************************************/
+/**
+ * Method returns the damping factor based on a sigmoid function on the value of each singular value.
+ */
+Eigen::MatrixXd DampingSigmoid::getDampingFactor(const Eigen::VectorXd& sorted_singular_values,
+                                                    const Eigen::MatrixXd& jacobian_data) const
+{
+    // Formula will be described in a future paper (to add reference)
+    double least_singular_value = sorted_singular_values(sorted_singular_values.rows() - 1);
+    uint32_t rows = sorted_singular_values.rows();
+    Eigen::MatrixXd damping_factor = Eigen::MatrixXd::Zero(rows,rows);
+    for(unsigned i = 0; i < sorted_singular_values.rows(); i++)
+    {
+      if (sorted_singular_values[i] < this->params_.eps_damping)
```

Comment:
ah, here you are already using ```eps_damping``` so only the description in the cfg-file is wrong....(you added it to ```eps_truncation```